### PR TITLE
Update did-auth spec with new definitions

### PR DESCRIPTION
**Pull request by javiesses**
- Update description
- Update sequence diagram

### CODE REVIEW

File: ssi/specs/did-auth.markdown [hunk 1]
```diff
@@ -6,27 +6,27 @@ layout: default
...
 
 1. Alice sends `POST /request_auth { did }` to Bob, where `did` is Alice's DID
-2. Bob creates a random challenge to send to Alice, stores the pair `did-challenge`, and responses with `{ challenge, sdreq }` were `sdreq` is the _Selective disclosure request_
-3. Alice receives Bob's challenge and selective disclosure request, obtains information required by the sdr, creates a JWT {% include ref.html id="1" %} with payload `{ challenge, sdres }` signed with `did`'s private key, where `sdres` is the selective disclosure response, and sends  `POST /auth { response: jwt({ challenge, sdres }) }`
-4. Bob verifies JWT signature, compares the challenge in the payload to the stored challenge, and performs business logic over the selective disclosure response. If is a valid user, creates and responses a Verifiable Credential that can be used by Alice to access the HTTPS service.
+2. Bob creates a random challenge to send to Alice, stores the pair `did-challenge`, and responses with `{ challenge, schemas }` were `schemas` is an array of published and well known names of _W3C Verifiable Credentials JSON Schema_. IE: `{ challenge: 'qwerty12345678', schemas: ['EmailCredentialSchema'] }`
```
Comment (ilanolkies) [inline ssi/specs/did-auth.markdown:22]:
```suggestion
2. Bob creates a random challenge to send to Alice, stores the pair `did-challenge`, and responses with `{ challenge, schemas }` were `schemas` is an array of published names of _W3C Verifiable Credentials JSON Schema_. IE: `{ challenge: 'qwerty12345678', schemas: ['EmailCredentialSchema'] }`
```

Needs to define what "published is"

Comment (ilanolkies) [reply ssi/specs/did-auth.markdown:22]: `schemas` i would change it for `sdr`, selective disclosure request. I would also make it more robust, specifying the sdr is asking for credentials
```
{
  challenge,
  sdr: {
    credentials: ['EmailCredential']
  }
}
```

**Review by ilanolkies — CHANGES_REQUESTED** [review]

File: ssi/specs/did-auth.markdown (hunk 1 above)
Comment (javiesses) [reply ssi/specs/did-auth.markdown:22]: I think this is a good option if we might add other stuff to the `sdr`, not just credentials. IE: `declarativeDetails`

Comment (ilanolkies) [reply ssi/specs/did-auth.markdown:22]: Let's check uPort DAF compatibility 😄

File: ssi/specs/did-auth.markdown [hunk 2]
```diff
@@ -6,27 +6,27 @@ layout: default
...
+3. Alice receives Bob's challenge and schemas request, obtains information required, creates a JWT {% include ref.html id="1" %} with payload `{ challenge, schemasResponse }` signed with `did`'s private key, where `schemasResponse` is the required schema response, and sends  `POST /auth { response: jwt({ challenge, schemaResponse }) }`
+4. Bob verifies JWT signature, compares the challenge in the payload to the stored challenge, and performs business logic over the schemas response. If is a valid user, creates and responses an access token. That access token is a `jwt({ issuer, subject, expiration, audience, schemaResponse })`.
 5. Alice accesses next HTTPS requests using header `Authentication` with the Credential received
```
Comment (ilanolkies) [inline ssi/specs/did-auth.markdown:25]: What about cookies?

**Pull request merged**